eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -356,20 +349,15 @@ instance
   where
   emptyImpNES = emptyShelleyImpNES
 
-  impWitsVKeyNeeded = shelleyImpWitsVKeyNeeded
-
-shelleyImpWitsVKeyNeeded ::
-  ( ProtVerAtMost era 8
-  , EraTx era
-  , ShelleyEraTxBody era
-  ) =>
+impWitsVKeyNeeded ::
+  EraUTxO era =>
   NewEpochState era ->
   TxBody era ->
   Set.Set (KeyHash 'Witness (EraCrypto era))
-shelleyImpWitsVKeyNeeded nes txb = shelleyWitsVKeyNeeded utxo txb genDelegs
+impWitsVKeyNeeded nes txBody = getWitsVKeyNeeded certState utxo txBody
```

Comment:
I wonder if this function is even necessary, it seems like it served the same purpose as the era-agnostic `getWitsVKeyNeeded` and doesn't do much besides lens-ing. 
But it's not important for this PR, just a thought.

Reply:
Yeah, I didn't really check how it is used if at all. From a quick grep it indeed looks like we can get rid of it